File: LeadToOffer/01_FindNumInMatrix.cpp
//1.cpp version
class Solution {
public:
    bool Find(int target, vector<vector<int> > array) {
        int n = array.size();
        if (array[0].size() == 0) {
            return false;
        }

        int row = 0;
        int col = n-1;
        while (row < n && col >= 0) {
            if (array[row][col] == target) {
                return true;
            } else if (array[row][col] < target) {
                row ++;
            } else {
                col --;
            }
        }
        return false;
    }
};

File: LeadToOffer/01_FindNumInMatrix.py
//1. search in acsending matrix
//First try to start from left top but not work.
//For right top is then ok

class Solution:
    def Find(self, target, array):
        n = len(array)
        if n == 0 or len(array[0]) == 0:
            return False
        i, j = 0, n-1
        while i < n and j >= 0:
            if array[i][j] == target:
                return True
            elif array[i][j] < target:
                i += 1
            else:
                j -= 1
        return False
